mkosi/config.py

```diff
@@ -5244,6 +5245,35 @@ def finalize_term() -> str:
     return term if sys.stderr.isatty() else "dumb"
 
 
+def finalize_git_config(proxy_url: Optional[str], env: dict[str, str]) -> dict[str, str]:
+    if proxy_url is None:
+        return {}
+
+    cnt_str = env.get("GIT_CONFIG_COUNT", "0")
+    if not cnt_str.isdigit():
+        raise ValueError("GIT_CONFIG_COUNT environment variable must be set to a valid integer")
+    cnt = int(cnt_str)
+
+    env_override = {}
+    for i in range(cnt):
+        key = env.get(f"GIT_CONFIG_KEY_{i}")
+        if key is None:
+            raise ValueError(
+                f"GIT_CONFIG_KEY_{i} environment variable must be set with GIT_CONFIG_COUNT={cnt}"
+            )
```

Comment:
What does git do here? i.e. is git fine if `GIT_CONFIG_COUNT` is nonzero but these variables are not set? If so, we shouldn't mess with these semantics. Maybe we shouldn't check for this, since we don't mess with these variables outside of this?

Reply:
git will emit an error per docs: https://git-scm.com/docs/git-config#Documentation/git-config.txt-GITCONFIGCOUNT

>  Any missing key or value is treated as an error. 

Given below where this whole block is unneccessary, yes lets remove this check

Reply:
Fixed in newest version